src/datachain/catalog/catalog.py

```diff
@@ -97,6 +98,46 @@ def noop(_: str):
     pass
 
 
+class TerminationSignal(RuntimeError):  # noqa: N818
+    def __init__(self, signal):
+        self.signal = signal
+        super().__init__("Received termination signal", signal)
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.signal})"
+
+
+if sys.platform == "win32":
+    SIGINT = signal.CTRL_C_EVENT
+else:
+    SIGINT = signal.SIGINT
+
+
+def shutdown_process(
```

Comment:
unit test this?

Reply:
There's no point in unittesting this. We'll check `mock_popen.terminate.assert_called_once()` which is not that useful.

Reply:
I mean function test this - it should not be about testing "mocks" or course